pyOCD/flash/flash.py

```diff
@@ -105,20 +109,42 @@ def __init__(self, target, flash_algo):
     def minimumProgramLength(self):
         return self.min_program_length
 
-    def init(self):
+    def init(self, adr=None, clk=0, fnc=None):
         """
         Download the flash algorithm in RAM
         """
+        if adr is None:
+            adr = self.getFlashInfo().rom_start 
+        
+        assert fnc in self.INIT_FUNCTIONS
+        
         self.target.halt()
         self.target.setTargetState("PROGRAM")
 
         # update core register to execute the init subroutine
-        result = self.callFunctionAndWait(self.flash_algo['pc_init'], init=True)
+        result = self.callFunctionAndWait(self.flash_algo['pc_init'], 
+                                          r0=adr, r1=clk, r2=fnc, init=True)
 
         # check the return code
         if result != 0:
             logging.error('init error: %i', result)
 
+    def unInit(self, fnc):
```

Comment:
Can you change "unInit" to "uninit" (single word, no camel-case)?

Reply:
fixed